spec/support/temp_aws_env.rb

```diff
@@ -126,16 +126,22 @@ def self.switch_to_temporary_credentials
 
       ENV['VCR_IGNORE_REQUESTS_TEMPORARILY'] = 'true'
 
-      ENV['AWS_MY_ARN'] = Aws::IAM::CurrentUser.new(region: "us-east-1").arn
-      filter_env_var('AWS_MY_ARN')
-
       sts_client = Aws::STS::Client.new(region: "us-east-1")
-      resp = sts_client.get_session_token({duration_seconds: 3600})
-      @@temporary_credentials = resp.credentials
+      identity = sts_client.get_caller_identity
+      already_in_temporary_role = identity.arn.match?(/:assumed-role\//)
+
+      if !already_in_temporary_role
```

Comment:
we didn't use to have aws-access and temporary credentials; if we are using aws-access skip the getting of new temporary credentials (AWS won't let us get temporary credentials from temporary credentials).